src/app/tests/TestReadInteraction.cpp

```diff
@@ -1632,6 +1632,10 @@ void TestReadInteraction::TestSubscribeRoundtrip(nlTestSuite * apSuite, void * a
         delegate.mGotReport            = false;
         delegate.mNumAttributeResponse = 0;
 
+        // Make sure the reporting engine actually runs.
+        err = engine->GetReportingEngine().SetDirty(dirtyPath4);
```

Comment:
The test is trying to test that the report that runs after the `DrainAndServiceIO` call on line 1635 (before this change, 1639 after) happens does not have any attributes in it.

Before this change, after line 1624 the ReadHandler is not reportable, then we make it reportable _but nothing schedules a reporting run_.  So the test passed because there was no report sent at all.  It wasn't testing what it meant to test.

This change was trying to make it so a reporting run at least happens, but I think it was made before I finished making the various ReadHandler changes, so I agree that in its current form it doesn't actually do anything....  I will remove this bit and do a separate PR to make this test actually meaningful.

Reply:
Removed this change, filed a followup and added a comment pointing to the followup.